### ISSUE TITLE
Parentheses are added to XML even inside String literals (0.13.7)

### ISSUE THREAD

**Issue by pfn** [sbt/sbt#1738]
```
TaskKey[Unit]("crashlytics-gen") <<= (collectResources in Android) map {
  case (_,res) =>
  val resource = """<string name="com.crashlytics.android.build_id">%s</string>""" format java.util.UUID.randomUUID.toString
  val lines = """<?xml version="1.0" encoding="utf-8"?>""" ::
    "<resources>" :: resource :: "</resources>" :: Nil
  println("wtf: " + lines.mkString(""))
  lines foreach println
}
```

Produces this:

```
> crashlytics-gen
wtf: <?xml version="1.0" encoding="utf-8"?>(<resources>(<string name="com.crashlytics.android.build_id">f7a7b0d5-9ac9-4c1c-97c7-2fb852fa5bfc</string>)</resources>)
<?xml version="1.0" encoding="utf-8"?>
(<resources>
(<string name="com.crashlytics.android.build_id">f7a7b0d5-9ac9-4c1c-97c7-2fb852fa5bfc</string>)
</resources>)
```

**Comment by pfn** [comment]
More madness

```
[pfnguyen@GALACTICA t] $ cat build.sbt; sbt run
run := {
  val resource = """<string name="com.crashlytics.android.build_id">%s</string>""" format java.util.UUID.randomUUID.toString
  val lines = s"""
    |<?xml version="1.0" encoding="utf-8"?>
    |<resources>
    |  $resource,
    |</resources>
  """.stripMargin
  println("wtf: " + lines)
}

[info] Loading global plugins from C:\Users\pfnguyen\.sbt\0.13\plugins
[info] Loading project definition from C:\Users\pfnguyen\t\project
[info] Set current project to t (in build file:/C:/Users/pfnguyen/t/)
wtf:
<?xml version="1.0" encoding="utf-8"?>
(<resources>
  (<string name="com.crashlytics.android.build_id">86f3f923-99e7-4b5a-99bc-c6a64a5539e6</string>),
</resources>)

[success] Total time: 0 s, completed Nov 19, 2014 2:16:58 PM
```

**Comment by pfn** [comment]
Using native xml tags avoids this problem...

**Comment by eed3si9n** [comment]
Likely related to code around #1671 /cc @ajozwik

**Comment by ajozwik** [comment]
See #1606.
Scala compiler has problem - if after xml is literal (or any other statement). As work-around sbt parser adds to "xml parts" brackets. It has to check - if xml code is string - like in your example. Parsing sbt file start to be more complicated - than we expected.

**Comment by jsuereth** [comment]
@ajozwik I think perhaps we need to start investigating feeding the Trees out of the parser directly to `Eval` in some fashion, and reworking the SessionSetting refactoring code against Trees.   maybe we should do a conference call to discuss options here, but for sbt 1.0 we CAN break binary compatibility, so let's figure out if a better architecture will lead to less issues.

**Comment by ajozwik** [comment]
@jsuereth You right. From my side I do not have enough knowledge about sbt. The problem with xml is in scala compiler - not sbt. We have two options:
1. Correct the work-around for xml (I will create new PR for this).
2. Write migration document and remove `XmlContent.handleXmlContent` method.

I think the first option is good for now. For sbt 1.0 the second should be preferred.

If you think global about parsing sbt files - I think the output should be Trees. But this is not one month work after work :).

**Comment by ajozwik** [comment]
@jsuereth @eed3si9n @gkossakowski @WarsawScala @jaceklaskowski @rkrzewski: We can arrange the conference call 3 December at 18:30 CET (12:30 GMT-5) during our next meetup (http://www.meetup.com/WarszawScaLa). 
Please confirm.

**Comment by ajozwik** [comment]
Please give me a message, when do you prefer to arrange the conference - we have a meeting every two weeks. If not 3 December - maybe next (17 December) or in new year?

**Comment by gkossakowski** [comment]
Hi @ajzowik! I missed your question before.

I could attend meeting at Dec 3rd at 18:30 but for maximum an hour. I have some other meetings later on.

**Comment by jsuereth** [comment]
Unfortunately I have a conflict the 3rd.  Let me see if I can reschedule

**Comment by eed3si9n** [comment]
I'm available Dec 3rd.

**Comment by jsuereth** [comment]
Ok, I'm free on the 3rd.  Let's do a G+/skype call.  You can email us connection details.

**Comment by ajozwik** [comment]
So let's talk tomorrow.

**Comment by gkossakowski** [comment]
Unfortunately, something else popped up so I won't be able to do the video call at 18.30 CET. How about 17.00 CET?

**Comment by ajozwik** [comment]
We have a meeting room from 18:00 CET. Most people arrive between 18:00 and 18:30.

**Comment by gkossakowski** [comment]
I see. I'll need to pass then, unfortunately.

**Comment by ajozwik** [comment]
The meetup announcement:
http://www.meetup.com/WarszawScaLa/events/219028504/ 
I have slackware on my laptop - I tested G+

**Comment by rkrzewski** [comment]
During the call we found out that contrary to our earlier belief the problem is not caused by a particular way we use the parser provided by scala-reflect Toolbox, but exists in the parser used by scalac itself.

Here's the Jira issue with minimized test case: https://issues.scala-lang.org/browse/SI-9027

**Comment by rkrzewski** [comment]
Aaand it's fixed! https://github.com/scala/scala/pull/4185 
We need a backport of this change to Scala 2.10 though.
/cc @jsuereth @eed3si9n

**Comment by eed3si9n** [comment]
I believe this is fixed via #1765.
